### ISSUE TITLE
disable systemd-resolved

### ISSUE THREAD

**Issue by mamama1** [nextcloud/vm#1372]
Hi

I'd like to propose disabling systemd-resolved.

Disclaimer: A rant follows, I may regret later and/or not mean things 100% like I'll say them in the rant. Beware, read on your own risk. Scroll past the horizontal line to skip it.

This idiotic piece of s*** called systemd-resolved seems to treat local domains which end with ".local" differently which leads to non-working name resolution. The devs of that piece of junk took something that was working perfectly for decades and broke it due to missing knowledge and not thinking outside their (very small) box (.local is reserved for multicast BLAH). WHO needs this retarded steaming pile of dog turd listening locally on port 53 and messing with DNS requests until they return nothing but utter BS?

100% ack to this (found on the internet): `Launching a local DNS server on port 53 is just another example of systemd hubris and is an unbelievable example of how they do not know what they are doing.`

-------------------------------------------------------------

"ping anyserver.mydomain.local" doesn't resolve, thus for example LDAP is not configurable in nextcloud. As usind the ".local" suffix for local Active Directory domains has been best practice for decades(?) in the Windows world, many users may run into this issue.

I vote therefore for:

```
sudo systemctl disable systemd-resolved.service
sudo systemctl stop systemd-resolved
rm /etc/resolv.conf (remove the symlink)
```

and recreate resolv.conf with the dns servers which were supplied in the setup script.
Why does every as simple as braindead task has to be made utterly complicated using ridiculous constructions like systemd-resolved. Jeeeeeeeeeeez, I know why I normally wouldn't touch Ubuntu with a pair of pincers...

**Comment by enoch85** [comment]
OK, so what are the alternative?

https://github.com/nextcloud/vm/blob/master/lib.sh#L55-L75

https://github.com/nextcloud/vm/blob/master/network/static_ip.sh

**Comment by mamama1** [comment]
Not sure, what do you mean? What's wrong with just filling /etc/resolv.conf with the user provided dns servers (or default dns servers if the user chooses to not provide any servers)?

**Comment by enoch85** [comment]
We recently fixed DNS issues in the VM, and if you run the static IP script, you DNS should be correct.

Please confirm.

**Comment by szaimen** [comment]
Maybe DNS should be a separate script from stati_ip inside server_configuration?

**Comment by mamama1** [comment]
the issue exists in the yesterday deployed VM, and I have run the static ip script.
execute a "ping myhostname.mydomain.local" and it will not resolve.
ping some other name which doesn't have ".local" as TLD (e.g. myhostname.mydomain.net), I get the correct result from my private DNS server. How do I know that the response comes from my prviate DNS server? Well, I have a split brain configuration where my private DNS resolves the same name to a different (local) ip address than a public DNS server and the appliance resolves the ping request to the local ip address. So it is verified, that dns requests are being made to my private DNS servers which i supplied in the static ip script. At least if the domain doesn't end with .local.

**Comment by enoch85** [comment]
Well, I can't answer for super custom configurations. But I know for a fact that LDAP works with local DNS. I've set it up several times before.

You need to use your local DNS servers for it to work.

**Comment by mamama1** [comment]
As I said, i have provided my local DNS servers (which are my Active Directory Domain Controllers) during the setup, during setting the static IP.

Trying to ping mydomaincontroller.mydomain.local fails, so does dig and nslookup as well.
Any domains and host names which don't end with ".local" resolve without an issue.

Like mentioned before, this seems to be due to the fact that the systemd-resolved people are saying that .local is reserved for some multicast m-dns stuff, not taking into account that .local has been used for decades for Active Directory deployments.

As soon as I obeyed systemd-resolved by disabling the service and recreating /etc/resolv.conf, filling it with my local DNS servers, .local name resolution started to work again.

Are there any downsides from your point of view to this configuration? I read somewhere that docker contains might have DNS issues but so far even Bitwarden is still able to send mails, which requires working name resolution in my case.

**Comment by enoch85** [comment]
Yeah, I read that .local didn't work for you, but I managed to get it to work before - though it was like 1-2 years ago. :) What strikes me as odd, is that not much has chanhed since then. We rather improved stuff over the years..

I'm 100% Linux, so I don't use LDAP myself - hence I never tried .local domains.

Maybe we can work around it though script wise. Like if .local is used, then disable systemd-resolved. Any suggestions?

**Comment by enoch85** [comment]
Is this still valid @mamama1?

**Comment by enoch85** [comment]
Sorry, but this will likely never happen.
